### PR TITLE
NVMe Scaling Optimization

### PR DESCRIPTION
**Pull request by harshitap26**
# Description
PR for optimization of finding the nvme devices.

Dependency - Need to add one more commit after the gonvme PR is merged to update the go.mod file - https://github.com/dell/gonvme/pull/18/

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/351|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x]  Tested on linux environment with nvme library installed
- [x] NVMe pods go to running state with these changes
- [x] The time for scaling suite has reduced as expected and we are able to create 20 replicas in around 4 mins

### CODE REVIEW

**Review by harishp8889 — APPROVED** [review]

File: go.sum [hunk 1]
```diff
@@ -5,6 +5,10 @@ github.com/dell/goiscsi v1.4.0 h1:TpMo0BLQpHsuUaePHBwZ0yjcq8gXhOOuYeUPzqmwjKk=
 github.com/dell/goiscsi v1.4.0/go.mod h1:oDkUXDPAYqSOc0t6PpU+BpOqF45mzv+9eLFRkAoQYKs=
 github.com/dell/gonvme v1.1.0 h1:5O554dUsrPEXNEy/mc+977eD1FE0geLYlk5HwuXnAaE=
 github.com/dell/gonvme v1.1.0/go.mod h1:qLlZNduxrt1hLR/JaU6OXe8+QkU6dtwa5wHM8EWQ58Y=
+github.com/dell/gonvme v1.1.1-0.20220629121439-4a54a8073e7d h1:Cdbh1hWGoEnbnn0yqnzrLUQuzsXoNhUhXcR3HEt5HiE=
+github.com/dell/gonvme v1.1.1-0.20220629121439-4a54a8073e7d/go.mod h1:qLlZNduxrt1hLR/JaU6OXe8+QkU6dtwa5wHM8EWQ58Y=
+github.com/dell/gonvme v1.1.1-0.20220704070618-c22e60197ee5 h1:9qo+9y4LrLCvKsUoAUkglO/GfvbLWczaePlkwbQPAJs=
+github.com/dell/gonvme v1.1.1-0.20220704070618-c22e60197ee5/go.mod h1:qLlZNduxrt1hLR/JaU6OXe8+QkU6dtwa5wHM8EWQ58Y=
```
Comment (spriya-m) [inline go.sum:11]: go mod tidy needed here

File: base.go [hunk 2]
```diff
@@ -124,7 +124,43 @@ func (bc *baseConnector) disconnectNVMEDevicesByDeviceName(ctx context.Context,
...
+	for _, d := range newDevices {
+		err := bc.scsi.DeleteSCSIDeviceByName(ctx, d)
+		if err != nil {
+			logger.Error(ctx, "can't delete block device: %s", err.Error())
+			if !force {
+				return err
+			}
+		}
+		if dm != "" {
+			_ = bc.multipath.DelPath(ctx, path.Join("/dev", d))
```
Comment (spriya-m) [inline base.go:160]: error handling not needed for DelPath()?

Comment (harshitap26) [reply base.go:160]: We have implemented the same way as done for ISCSI. Also we cannot find this path for nvme, so I guess that's why error handling is not needed. Its just cleaning the nvme devices.

Comment (harshitap26) [reply base.go:160]: Error handling is not required. The error case implies the device is not there which means it had been cleaned up

Comment (spriya-m) [reply base.go:160]: ok

**Review by spriya-m — APPROVED** [review]

**Pull request merged**